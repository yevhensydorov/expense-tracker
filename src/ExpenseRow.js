import React from 'react';

class ExpenseRow extends React.Component {

	render() {
		const oneExpense = this.props.expense;
		console.log('come from expense row', oneExpense)
		// let row = oneExpense.map((item, index) => {
		// 	// return 
		// 	// 	<tr key={index}>
		// 	// 		<td>item.description</td>
		// 	// 		<td>item.date</td>
		// 	// 		<td>item.amount</td>
		// 	// 	</tr>
		// 	// console.log(item)
		// });

		return (
			<tbody>
				{/*{row}*/}
			</tbody>
		);
	}
}

export default ExpenseRow;