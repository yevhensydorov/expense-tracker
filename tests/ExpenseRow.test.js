import React from "react";
import ExpenseRow from "../src/ExpenseRow";
import renderer from "react-test-renderer";

const dummyExpenses = [
  { description: "cheese", date: "2017-11-30", amount: "3243" },
  { description: "wine", date: "2018-11-21", amount: "32" },
  { description: "strawberry", date: "2018-12-05", amount: "3" }
];

describe("ExpenseRow", () => {
  it("matches the snapshot", () => {
    const tree = renderer
      .create(<ExpenseRow expense={dummyExpenses} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
