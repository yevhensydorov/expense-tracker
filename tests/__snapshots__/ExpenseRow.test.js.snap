// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseRow matches the snapshot 1`] = `
<tbody>
  <tr>
    <td>
      cheese
    </td>
    <td>
      2017-11-30
    </td>
    <td>
      3243
    </td>
  </tr>
  <tr>
    <td>
      wine
    </td>
    <td>
      2018-11-21
    </td>
    <td>
      32
    </td>
  </tr>
  <tr>
    <td>
      strawberry
    </td>
    <td>
      2018-12-05
    </td>
    <td>
      3
    </td>
  </tr>
</tbody>
`;
